src/Tooltip/Tooltip.component.tsx

```diff
@@ -0,0 +1,262 @@
+                    tooltipMargin}px, 0)`,
+            });
+        case 'rightStart':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(${width + tooltipMargin}px, 0, 0)`,
+            });
+        case 'right':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(${width + tooltipMargin}px, ${height /
+                    2 -
+                    toolTipHeight / 2}px, 0)`,
+            });
+        case 'rightEnd':
+            return css({
+                left: 0,
+                bottom: 0,
+                transform: `translate3d(${width + tooltipMargin}px, 0, 0)`,
+            });
+        case 'bottomEnd':
+            return css({
+                right: 0,
+                top: 0,
+                transform: `translate3d(0, ${height + tooltipMargin}px, 0)`,
+            });
+        case 'bottom':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(${width / 2 -
+                    toolTipWidth / 2}px, ${height + tooltipMargin}px, 0)`,
+            });
+        case 'bottomStart':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(0, ${height + tooltipMargin}px, 0)`,
+            });
+        case 'leftEnd':
+            return css({
+                left: 0,
+                bottom: 0,
+                transform: `translate3d(-${toolTipWidth +
+                    tooltipMargin}px, 0, 0)`,
+            });
+        case 'left':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(-${toolTipWidth +
+                    tooltipMargin}px, ${height / 2 - toolTipHeight / 2}px, 0)`,
+            });
+        case 'leftStart':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(-${toolTipWidth +
+                    tooltipMargin}px, 0, 0)`,
+            });
+        default:
+            return css({
+                right: 0,
+                top: 0,
+                transform: `translate3d(0, -${height + tooltipMargin}px, 0)`,
+            });
+    }
+};
+
+const TooltipElement = styled('div')<TooltipElementProps>`
+    position: absolute;
+    box-sizing: border-box;
+    ${({ background = 'rgba(0, 0, 0, 0.8)', color = 'white' }) =>
+        css({ background, color })};
+    border-radius: base;
+    font-family: ${th('typography.fontFamily')};
+    font-size: 0.8rem;
+    padding: 0.5rem;
+    opacity: 0;
+    visibility: hidden;
+    transition: opacity 250ms ease-in-out;
+
+    &.active {
+        opacity: 1;
+        visibility: visible;
+    }
+
+    ${({ wrap = true }) =>
+        wrap ? css({ whiteSpace: 'normal' }) : css({ whiteSpace: 'nowrap' })};
+
+    ${({ maxWidth = 'auto' }) => css({ width: maxWidth })};
```

Comment:
applying `maxWidth` to `width` style?

Reply:
`maxWidth` is a prop for defining the tooltip width. Width is a prop that we use for passing the width of the rendered tooltip around. I was going to change the width prop, but maxWidth seems more semantic.